connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -132,25 +132,36 @@ public DelegatingClassLoader(List<String> pluginPaths) {
         return connectorClientConfigPolicies;
     }
 
+    public PluginClassLoader pluginClassLoader(String name) {
+        if (PluginUtils.shouldLoadInIsolation(name)) {
```

Comment:
Nit: to avoid nesting, I'd personally go with:
```suggestion
        if (!PluginUtils.shouldLoadInIsolation(name)) {
            return null;
        }
```

Even if it does lead to two `return null` statements, they arise from different circumstances, so the duplication there isn't a problem.

Reply:
I'm going to refactor this into three different `return null`s for completeness.